ledger/src/sigverify_shreds.rs

```diff
@@ -102,264 +82,9 @@ fn verify_shreds_cpu(
-            .map(move |offset| Some(offset? + shift))
-    };
-    let offset = pubkeys.len() + merkle_roots.len();
-    let (signature_offsets, msg_start_offsets, msg_sizes) =
-        shred_gpu_offsets(offset, batches, merkle_roots_offsets, recycler_cache);
-    let mut out = recycler_cache.buffer().allocate("out_buffer");
-    out.set_pinnable();
-    out.resize(signature_offsets.len(), 0u8);
-    let mut elems = vec![
-        elems_from_buffer(&pubkeys),
-        elems_from_buffer(&merkle_roots),
-    ];
-    // `BytesPacketBatch` cannot be directly used in CUDA. We have to retrieve
-    // and convert byte batches to pinned batches. We must collect here so that
-    // we keep the batches created by `BytesPacketBatch::to_pinned_packet_batch()`
-    // alive.
-    let pinned_batches = batches
-        .iter()
-        .map(|batch| match batch {
-            PacketBatch::Pinned(batch) => Cow::Borrowed(batch),
-            PacketBatch::Bytes(batch) => Cow::Owned(batch.to_pinned_packet_batch()),
-            PacketBatch::Single(packet) => {
-                // this is ugly, but unused (gpu code) and will be removed shortly in follow up PR
-                let mut batch = BytesPacketBatch::with_capacity(1);
-                batch.push(packet.clone());
-                Cow::Owned(batch.to_pinned_packet_batch())
-            }
-        })
-        .collect::<Vec<_>>();
-    elems.extend(pinned_batches.iter().map(|batch| perf_libs::Elems {
-        elems: batch.as_ptr().cast::<u8>(),
-        num: batch.len() as u32,
-    }));
-    let num_packets = elems.iter().map(|elem| elem.num).sum();
-    trace!("Starting verify num packets: {num_packets}");
-    trace!("elem len: {}", elems.len() as u32);
-    trace!("packet sizeof: {}", size_of::<Packet>() as u32);
-    const USE_NON_DEFAULT_STREAM: u8 = 1;
-    unsafe {
-        let res = (api.ed25519_verify_many)(
-            elems.as_ptr(),
-            elems.len() as u32,
-            size_of::<Packet>() as u32,
-            num_packets,
-            signature_offsets.len() as u32,
-            msg_sizes.as_ptr(),
-            pubkey_offsets.as_ptr(),
-            signature_offsets.as_ptr(),
-            msg_start_offsets.as_ptr(),
-            out.as_mut_ptr(),
-            USE_NON_DEFAULT_STREAM,
-        );
-        if res != 0 {
-            trace!("RETURN!!!: {res}");
-        }
-    }
-    trace!("done verify");
-    trace!("out buf {out:?}");
-
-    // Each shred has exactly one signature.
-    let v_sig_lens = batches
-        .iter()
-        .map(|batch| iter::repeat_n(1u32, batch.len()));
-    let mut rvs: Vec<_> = batches.iter().map(|batch| vec![0u8; batch.len()]).collect();
-    sigverify::copy_return_values(v_sig_lens, &out, &mut rvs);
-
-    inc_new_counter_debug!("ed25519_shred_verify_gpu", out.len());
-    rvs
-}
-
 #[cfg(test)]
 fn sign_shred_cpu(keypair: &Keypair, packet: &mut PacketRefMut) {
-    let sig = shred::layout::get_signature_range();
+    let sig = 0..64;
```

Comment:
Why this change?

Reply:
To declare test-only functions that are used across crates it is possible to use dev-context-only-utils. Not important here though.